onedal/common/_policy.py

```diff
@@ -46,33 +42,19 @@ def _get_queue(*data):
     return None
 
 
-class _Daal4PyContextReset:
-    def __init__(self):
-        self._d4p_context = None
-        if oneapi_is_available:
-            self._d4p_context = _get_sycl_ctxt()
-
-    def __del__(self):
-        if self._d4p_context:
-            self._d4p_context.apply()
-
-
 class _HostInteropPolicy(_backend.host_policy):
     def __init__(self):
         super().__init__()
-        self._d4p_interop = _Daal4PyContextReset()
 
 
 if _is_dpc_backend:
 
     class _DataParallelInteropPolicy(_backend.data_parallel_policy):
         def __init__(self, queue):
             self._queue = queue
-            self._d4p_interop = _Daal4PyContextReset()
-            if "sklearnex" in sys.modules:
-                from sklearnex._device_offload import DummySyclQueue
+            from onedal._device_offload import DummySyclQueue
```

Comment:
We don't have any init file for policies imports and it seems to me that this import should be isolated, since in the policy module I do not want to add unnecessary things for import.

If there are any suggestions for the refactoring, we can create tickets for that.

I have already shared perf validations, no regressions here.

Reply:
I don't see any reason why it should not be directly imported after line 50 `if _is_dpc_backend:`